EIPS/eip-2645.md

```diff
@@ -0,0 +1,68 @@
+---
+eip: 2645
+title: Hierarchical Deterministic Wallet for Computation Integrity Proof (CIP) Layer-2
+author: Tom Brand (tom@starkware.co), Louis Guthmann (louis@starkware.co)
+discussions-to: https://ethereum-magicians.org/t/hierarchical-deterministic-wallet-for-computation-integrity-proof-cip-layer-2/4286
+status: Draft
+type: Standards Track 
+category: ERC
+created: 2020-05-13
+---
+
+## Simple Summary
+In the context of Computation Integrity Proof (CIP) Layer-2 solutions such as ZK-Rollups, users are required to sign messages on new elliptic curves optimized for those environnements. These curves are Baby Jubjub (in the context of the BN254 SNARK approach) and Arya in (the context of StarkEx). Extensive work has been done to provide secure ways to derive private keys. We leverage this work to define an efficient way to securely derive private keys from existing infrastructure, as well as creating domain separation between Layer-2 applications.
+
+## Abstract
+We provide a Derivation Path allowing a user to derive hierarchical keys for Layer-2 solutions depending on the zk-technology, the application, the user’s Layer-1 address, as well as an efficient grinding method to enforce the private key distribution within the curve domain.
```

Comment:
Consider including the derivation path here.  Abstract should be a human readable and very terse version of the specification.  Ideally, someone can read the abstract and get the gist of the specification without having to read anything else.  Usually minor details and edge cases are ignored, and only the core of the specification is included in plain English.  This abstract is pretty solid as is, would just be nice to throw `m / purpose' / layer' / application' / eth_address_1' / eth_address_2' / index` in there somewhere since that is the core of the specification.

Reply:
Done